packages/trpc/server/routers/viewer/eventTypes/__tests__/util.test.ts

```diff
@@ -0,0 +1,516 @@
+import { describe, it, expect, vi, beforeEach, type Mock } from "vitest";
+
+import { PermissionCheckService } from "@calcom/features/pbac/services/permission-check.service";
+import type { PrismaClient } from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+
+import { TRPCError } from "@trpc/server";
+
+import type { authedProcedure } from "../../../procedures/authedProcedure";
+import { createEventPbacProcedure } from "../util";
+
+// Mock dependencies
+vi.mock("@calcom/features/pbac/services/permission-check.service");
+
+describe("createEventPbacProcedure", () => {
+  let mockPermissionCheckService: {
+    checkPermission: Mock;
+  };
+
+  const mockPrisma = {
+    eventType: {
+      findUnique: vi.fn(),
+    },
+  } as unknown as PrismaClient;
+
+  const mockCtx = {
+    user: { id: 1, profile: { upId: "user-1" } },
+    session: { user: { id: 1 } },
+    prisma: mockPrisma,
+  };
+
+  const mockNext = vi.fn().mockResolvedValue({ ctx: mockCtx });
+
+  // Helper to get the custom middleware (after authedProcedure)
+  const getMiddleware = (procedure: ReturnType<typeof authedProcedure>) => {
+    // The last middleware is our custom one
+    return procedure._def.middlewares[procedure._def.middlewares.length - 1];
+  };
```

Comment:
I didnt know you can test trpc middleware like this until i found an example in someones git repo :D